watchtower/lookout/interface.go

```diff
@@ -0,0 +1,77 @@
+package lookout
+
+import (
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/btcsuite/btcd/wire"
+	"github.com/lightningnetwork/lnd/chainntnfs"
+	"github.com/lightningnetwork/lnd/watchtower/wtdb"
+)
+
+// Service abstracts the lookout functionality, supporting the ability to start
+// and stop. All communication and actions are driven via the database or chain
+// events.
+type Service interface {
+	// Start safely starts up the Interface.
+	Start() error
+
+	// Stop safely stops the Interface.
+	Stop() error
+}
+
+// BlockFetcher supports the ability to fetch blocks.
+type BlockFetcher interface {
+	// FetchBlockByHash fetchs the block given the target block hash.
+	GetBlock(*chainhash.Hash) (*wire.MsgBlock, error)
+}
+
+// DB abstracts the required persistent calls expected by the lookout. DB
+// provides the ability to search for state updates that correspond to breach
+// transactions confirmed in a particular block.
+type DB interface {
+	// GetLastMatchedEpoch returns the last block epoch at which the tower
+	// performed a match. If no match has been done, a nil epoch will be
+	// returned.
+	GetLastMatchedEpoch() (*chainntnfs.BlockEpoch, error)
+
+	// FindMatches searches its database for any state updates matching the
+	// provided breach hints. To be correct, a breach hint should be
+	// provided for each transaction included at the provided block epoch.
+	// If any matches are found, they will be returned along with encrypted
+	// blobs so that justice can be exacted.
+	FindMatches(*chainntnfs.BlockEpoch, []wtdb.BreachHint) ([]wtdb.Match, error)
```

Comment:
This just a database query, there's no network I/O here. The breach hints passed in are constructed from the block, though they could be constructed from multiple blocks.

The block epoch is passed just so that the database can record the last processed block height, so after startup it knows where to begin searching. i'm thinking about just making this a separate db call though.